cpp/include/cudf/reshape.hpp

```diff
@@ -0,0 +1,41 @@
+/*
```

Comment:
I think "reshaping.hpp" would better match the file naming style in our headers.  They are "category" names.

Reply:
But we have many headers that don't follow this:
sort -> sorting
replace -> replacement
join -> joining

Reply:
fwiw, I like using non-plural root words for type + file names, esp when the root word is already a verb describing the operation. But as always, consistency should trump preference. Seems like there's some inconsistency here, though?

Reply:
`ing` does not make a word plural. :)